protostar/commands/test/testing_live_logger.py

```diff
@@ -93,9 +154,18 @@ def log(
                             progress_bar.update(1)
                             tests_left_n -= 1
                 finally:
-                    progress_bar.write("")
-                    progress_bar.clear()
-                    self.log_testing_summary(test_collector_result)
+                    if structured_format:
+                        print(
+                            json.dumps(
+                                self.get_structured_testing_summary(
+                                    test_collector_result
+                                )
+                            )
+                        )
```

Comment:
`print + jsons.dumps` shouldn't be used anywhere in this PR. That's responsibility of `JsonMessenger` which also ensures the output is ndjson friendly.